### PR TITLE
Add graphing + slider support 

### LINKED ISSUE #7: Use line graphs to summarise time-series config and results
Graphs are much easier quickly review than data tables. Taswira demonstrates this well with it's [change graph](https://github.com/moja-global/GCBM.Visualisation_Tool/blob/master/screenshot.png).

These ones look nice and they have a clear example of displaying local and API data: https://vue-chartjs.org/guide/#chart-with-api-data

### PR DESCRIPTION
**Pull request by shubhamkarande13**
# Add graphing + slider support 

## Description

Fixes #7 

New features:

1) Tested and added a new slider from Vuetify as a SFC, to be used for Point config. The earlier one was only compatible with Vue 3 and not Vue 2, and hence broken, rendering the FLINT Example config useless. (see #25)

Update: Slider fixed by @waridrox from his end.

2) (Point module) Finally able to access the state from a component. 

image

3) Was able to isolate the simulation results from the response from the API and store them as an array. 

image

Next step: Plotting the pool values!

[Check out this Brush chart!](https://apexcharts.com/javascript-chart-demos/line-charts/brush-chart/)

### CODE REVIEW

**Comment by shubhamkarande13** [comment]
Lots and lots and lots of trial and error! 😅

**Comment by aornugent** [comment]
This is great Shubham! 

If I understand correctly, the trick to displaying results from the endpoint was to save it in store.js and write a getter to access the data?. 

data-forge looks useful for reshaping data. Maybe it'll also help in #4? What's vuetify for - is there something we don't have in vue-material?

File: flint.ui/package.json [hunk 1]
```diff
@@ -1,34 +1,43 @@
...
   "version": "1.1.0",
-  "description": "This project aims to implement a user-friendly UI for FLINT and GCBM runs",
   "private": true,
-  "license": "MIT",
+  "description": "This project aims to implement a user-friendly UI for FLINT and GCBM runs",
   "scripts": {
     "serve": "vue-cli-service serve --open",
-    "dev": "npm run serve",
     "build": "vue-cli-service build",
+    "build-watch": "vue-cli-service build-watch",
```
Comment (aornugent) [inline flint.ui/package.json:9]: Is there a difference here between colons and dashes? It looks like - is used in package name and : is used to tag scripts.

File: flint.ui/src/components/Cards/CardInfoConfig.vue [hunk 2]
```diff
@@ -11,36 +11,42 @@
           <span class="font-semibold text-xl text-blueGray-700">
             {{ cardTitle }}
           </span>
-          <button @click="showConfirmRunModal()" class="w-full mt-4 block align-middle flex-initial bg-white hover:bg-black hover:text-white text-gray-800 font-semibold py-2 px-4 border border-gray-400 rounded shadow">
-           <i class="far fa-play-circle"></i> {{cardFunctionRun}}
+          <button
+            @click="showConfirmRunModal()"
+            class="w-full mt-4 block align-middle flex-initial bg-white hover:bg-black hover:text-white text-gray-800 font-semibold py-2 px-4 border border-gray-400 rounded shadow"
```
Comment (aornugent) [inline flint.ui/src/components/Cards/CardInfoConfig.vue:16]: General question, not specifically for this PR. Can we define style classes elsewhere to make the template code easier to read?

File: flint.ui/src/components/Chart.vue [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+    received_data() {
+      var dataset = this.$store.state.received_data;
+      var pool_1 = [],
+        pool_2 = [],
+        pool_3 = [],
+        simulation_step = [];
+
+      var lines = dataset.split("\n");
+      lines.splice(0, 4);
+      lines.splice(-4);
```
Comment (aornugent) [inline flint.ui/src/components/Chart.vue:41]: Leave a comment to explain what's happening here

Comment (shubhamkarande13) [reply flint.ui/src/components/Chart.vue:41]: `dataset` refers to the `response.data` from the API, which is saved in the state as `received_data`. The above code snippet removes the highlighted parts of the response in the screenshots.

image

image

Comment (aornugent) [reply flint.ui/src/components/Chart.vue:41]: Thanks! I meant in the code 😅
```
# remove header from response
```

**Comment by shubhamkarande13** [comment]
> This is great Shubham!
> 
> If I understand correctly, the trick to displaying results from the endpoint was to save it in store.js and write a getter to access the data?.

Thank you Andrew! 
Yes, exactly. I am still figuring this out, right now it is working through `computed()`.
> 
> data-forge looks useful for reshaping data. Maybe it'll also help in #4? What's vuetify for - is there something we don't have in vue-material?

Yes we could fix #4 with `data-forge`. It is a bit difficult to play with data in JS unlike in Python, as we don't have JS alternatives for `numpy` and `pandas`. I tried `danfo-js` too but sadly it does not support parsing input from a string. 

I was not able to find a slider component in vue-material. Slider components are required to set the initial values of the Pools in FLINT Example Configuration. Currently they're broken. 😥
Hence I had to introduce Vuetify. Now @waridrox can import the slider as an SFC and bind it to state to update the modified configuration for a FLINT run.

**Comment by aornugent** [comment]
Thank you for explaining.

> Hence I had to introduce Vuetify. Now @waridrox can import the slider as an SFC and bind it to state to update the modified configuration for a FLINT run.

*A tip*
If 's better to try and encapsulate one feature per PR. Sometimes you can't! But this sounds like a case where adding vuetify goes naturally with implementing the slider SFC.

Don't worry this time, but for future reference, this approach makes it easier to isolate what is happening in each set of changes. 🙏

**Comment by aornugent** [comment]
I wonder if defining a filter might help with data manipulation? 🤔 For example `{{received_data | parse}}`

https://vuejs.org/v2/guide/filters.html

**Pull request merged**